crates/emitter/src/ast_emitter.rs

```diff
@@ -129,27 +131,31 @@ impl<'alloc> AstEmitter<'alloc> {
     fn emit_if(&mut self, if_statement: &IfStatement) -> Result<(), EmitError> {
         self.emit_expression(&if_statement.test)?;
 
-        let offset_alternate = self.emit.bytecode_offset();
-        self.emit.if_eq(0);
+        let alternate_patch: JumpPatchEmitter = ForwardJumpEmitter {
```

Comment:
sorry for confusing example in the previous comment.
you don't have to specify the type in this assignment (Rust can infer the type).

so
```
let alternate_patch = ForwardJumpEmitter {
```